### DISCUSSION TITLE
Does IrScrutinizer 2.4.1 generate codes well?

### DISCUSSION THREAD

**Discussion by Gelip** [bengtmartensson/IrScrutinizer#532] [General]
Hi. I use IrScrutinizer to generate all codes 0-255 for RCA-38 protocol (NIKAI), addr device F (15) for THOMSON TV 22HR3234
Most buttons I found and work but a few cannot find, for example, **OK** and **Source**. The red LED in TV reacts (flashes) to each IR code 0-255
To send codes I use Arduino Nano and sketch .ino exported from IrSrutinizer in format Infrared4Arduino:
![my_remote](https://github.com/user-attachments/assets/449fc4bf-f25d-4484-b1fd-da8153bd3b31)
I found in this https://conture.by/iresp/IR_DATABASE_TV(STB).pdf (page 444) that THOMSON code for OK button (17) is **BFF4** NIKAI format (RCA-38)
How convert NIKAI BFF4 to RAW or other format?

**Comment by bengtmartensson** [comment]
If I may sum up: NIKAY appers to be the same protocol we know as RCA-38, with another parametrization. There are a number of working signals using RCA-38 with D=15 (also Rob's update in the JP1-forum). If I understand you correctly, you have tried all signals with D=15 and for all 256 different Fs, and none works as the key OK, right?  Is there any way you can generate (from a program or a remote) that signal and capture it? 

>  How convert NIKAI BFF4 to RAW or other format?

You have to ask the author of the document.

**Comment by Gelip** [comment]
[sketch_Arduino.zip](https://github.com/user-attachments/files/16575654/sketch_Arduino.zip)
OK, I found codes - IrScrutinizer generate all OK - one more time I slowly checked all codes with TV :-)
Now, I know how convert beetwen NIKAI and RAW or vice versa:
![NIKAI_IRP_A1](https://github.com/user-attachments/assets/a456dbe7-84e5-419a-86e0-6568b0f93dc7)
![NIKAI_IRP_A2](https://github.com/user-attachments/assets/32ed0492-3f8c-49b0-adfa-0b50fd8012ad)

All finded buttons for THOMSON TV 22HR3234 - RCA-38, device addr=15 (F)

> CMD 	Button name
> 
>   0	RED
>   2	FAV
>   6	EXIT
>   8	Menu
>  18	Picture preset
>  26	Guide
>  30	TEXT
>  39	Last source
>  42	Power On/Off
>  44	PR-
>  45	PR+
>  46	VOL+
>  47	VOL-
>  48	0
>  49	1
>  50	2
>  51	3
>  52	4
>  53	5
>  54	6
>  55	7
>  56	8
>  57	9
>  60	INFO
>  63	Mute
>  86	LEFT
>  87	RIGHT
>  88	DOWN
>  89	UP
>  97	LIST
> 163	Source list
> 174	USB
> 216	BLUE
> 228	YELLOW
> 232	GREEN
> 235	Source change
> 244	OK
